mycli/main.py

```diff
@@ -307,18 +309,27 @@ def connect(self, database='', user='', passwd='', host='', port='',
         passwd = passwd or cnf['password']
         charset = charset or cnf['default-character-set'] or 'utf8'
 
+        # Favor whichever local_infile option is set.
+        for local_infile_option in (local_infile, cnf['local-infile'], 
+                                    cnf['loose-local-infile'], False):
```

Comment:
I was wondering why the default values for the `local_infile` in the function args is not False. But this logic assumes that if it is not explicitly set to false then we set it to None or empty string so it doesn't prematurely break out of this loop.

I think that's the reasoning as I understand it.